### DISCUSSION TITLE
Always connecting to default project

### DISCUSSION THREAD

**Discussion by alalawy** [supabase/supabase#7126] [Questions] [answered]
why my supabase self hosted always connectiong to default project ?
![Screen Shot 2022-05-31 at 13 59 20](https://user-images.githubusercontent.com/32614903/171622309-99e1535b-4f5a-45ef-8178-63c21f876b29.png)

**Comment by gumu1** [comment, marked as answer]
Hi, just figured out the problem. Probably you are running it on a VPS or VM. Just edit the .env file into supabase/docker and change localhost with the ip of the machine. Reboot the docker and it should works fine.

**Reply by SamMakesThings** [reply]
Specific values to change in .env:
SITE_URL
API_EXTERNAL_URL
SUPABASE_PUBLIC_URL

I think it has to be http and not https? e.g. http://12.345.678.901:3000

Hope this is helpful to a future wanderer

**Comment by silentworks** [comment]
Default project is the only project you get with self hosted supabase.

**Reply by Aman-Aalam** [reply]
But shouldn't it at least connect?
It doesn't connect.

**Comment by Mehdi-YC** [comment]
hello i got the same , 
with http://my-ip:3000/api/projects/default/subscription giving me a 404 error

**Reply by alalawy** [reply]
change the 3000 port to 8000

**Reply by moracabanas** [reply]
8000 is the Kong port isn't it?

**Comment by Jbmanllr** [comment]
Having this problem right now.
Running on localhost with latest pull.
First issue was kong running on 8000, didn't start, changed for 8080, now project keep on loading and not connecting...

**Comment by lcampana69** [comment]
Same problem with Windows 11 22H2

**Reply by moracabanas** [reply]
UPDATE: For no reason, I just changed back to `localhost` settings and it just work. I deleted the deployment and volumes and refresh local dev deployment and it just work I do not understand what is happening.

For Windows users (I am on windows 11 22H2 too) 

I made a guess from the next threads:
https://github.com/supabase/supabase/discussions/5639
https://github.com/supabase/supabase/pull/5402

->  `API_EXTERNAL_URL` is a variable `supabase-studio` continer is using from inside to make API calls to `kong` container.  

Then `API_EXTERNAL_URL=http://localhost:8000` is not accesible from the supabase-studio container, you cannot ping localhost from docker container in windows. 

You should use then `host.docker.internal` instead `localhost` 

I illustrate this behaviour in the next diagram

![image](https://user-images.githubusercontent.com/4382333/208009852-a6328aec-43ce-42ae-9b09-4214edff9b24.png)


PLEASE correct me if I am wrong, but it is working 😅

**Reply by moracabanas** [reply]
This solutions should work on mac also, as `host.docker.internal` is also available.

**Comment by Aman-Aalam** [comment]
Same problem here as well, and changing localhost to server IP address didn't make any difference.
Anybody resolved it?

**Reply by redhahari** [reply]
I have the same issue, running on docker desktop wsl2 ubuntu on windows 11. 
In my case I'm still using localhost since I'm trying it in local

**Comment by globemediaofficial** [comment]
Same issue... I just started using the newest pull and studio won't load the "Default Project." This wasn't an issue before and I am using the same .env file as I was when it worked. The `SITE_URL` is set to `0.0.0.0:3000` and visiting that opens the studio but it's stuck on connecting to the project.

**Comment by Samsonov0** [comment]
The problem is still relevant. I run it on a vps, in .env localhost is replaced by the ip of the server. The endless connection to the default project does not disappear anywhere

**Comment by chit-uob** [comment]
I solve this problem by looking into Chrome development tools, inside the Network tab, it says it is trying to get domain/rest/v1:8000
Then I remembered that I configured Nginx to redirect normal domain/rest/v1:433 to :8000
So I went back to the .env, changed the API_EXTERNAL_URL, SUPABASE_PUBLIC_URL to the https://domain link, then it worked

**Comment by Valy-D** [comment]
Fixed. In case anyone needed another explanation
I have a Windows PC and I’m using supabase in a Linux-VM.

1) In the Linux-VM: go to ~/supabase/docker folder (like: cd ~/supabase/docker)

2) Edit .env file   (like: vi .env or nano .env)

3) !! I have changed these 2 lines that were pointed to localhost. When you connect from your "Windows" browser to supabase, localhost is your windows machine/pc and not your supabase within your Linux-VM. 
!! find these 2 lines and change their IP-ADDRESSes to the IP-ADDRESSes of the Server/PC/or VM that supabase is running. 
The ip-address of my server is 192.168.50.10

So, my changes were:
API_EXTERNAL_URL=http://localhost:8000  ! to !   API_EXTERNAL_URL=http://192.168.50.10:8000

And 
SUPABASE_PUBLIC_URL=http://localhost:8000    ! to !   SUPABASE_PUBLIC_URL=http://192.168.50.10:8000

**Comment by philmas** [comment]
To anyone who is still searching for a solution when setting up Supabase localhost. Try to reset (or check thoroughly) your environment variables / secrets. For me that appeared to be the problem. I had modified them incorrectly after which it started connecting indefinitely.

**Comment by godcodevo** [comment]
Hi, just figured out the problem. Probably you are running it on a VPS or VM. Just edit the .env file into supabase/docker and change localhost with the ip of the machine. Reboot the docker and it should works fine.
